kubejs/server_scripts/base/animation/animation_block.js

```diff
@@ -0,0 +1,185 @@
+      return animation_block_failure(event, "Name tag has not been renamed!")
+  
+    const name = heldItem.getHoverName().getString()
+    if(!isValidName(name))
+      return animation_block_failure(event, "Given name can only contain the following characters: a-z A-Z 0-9 _ . \- \/")
+    blockEntity.data.putString("name", name)
+    blockEntity.sync()
+
+    animation_block_setAmourStandName(event.server, blockEntity.getBlockPos(), name)
+  
+    return animation_block_success(event, Component.ofString("Name set to ").white().append(heldItem.getHoverName().italic()), true)
+  }  else if (heldItem.getItem() instanceof DyeItem)
+  { // Set outline color
+    let color
+    if (player.isShiftKeyDown())
+      color = heldItem.getItem().getDyeColor().getTextColor()
+    else
+      color = heldItem.getItem().getDyeColor().getTextColor()
+    blockEntity.data.putInt("color", color)
+    return animation_block_success(event, `Color set to ${heldItem.getItem().getDyeColor().getName().toUpperCase()}`)
+  } else if (heldItem.id.startsWith("ptdye:animation_anchor"))
+  {
+    if (heldItem.id == "ptdye:animation_anchor")
+      return animation_block_failure(event, "Anchor has no target set")
+    blockEntity.data.put("anchor", heldItem.nbt.get("target"))
+    return animation_block_success(event, `Anchor set to [${heldItem.nbt.getIntArray("target").join(' ,')}]`)
+  } else if(player.isShiftKeyDown())
+  { // Schematic
+    const name = blockEntity.data.getString("name")
+    if (!name)
+      return animation_block_failure(event, 
+        Component.ofString('No name set! You must use a ')
+                 .append(Component.of(Items.NAME_TAG.getName(Items.NAME_TAG)).aqua().underlined())
+                 .append(Component.ofString(' before you can save the structure!'))
+                 .white()
+        , true)
+    if(!isValidName(name))
+      return animation_block_failure(event, "Set name can only contain the following characters: a-z A-Z 0-9 _ . \- \/")
+
+    const min = blockEntity.data.getIntArray("min")
+    const max = blockEntity.data.getIntArray("max")
+    if (min.length < 3 || max.length < 3)
+      return animation_block_failure(event, "Cannot create structure from nothing. Check that there is a glued structure on one of the cardinal faces.")
+    let [minX, minY, minZ] = min
+    let [maxX, maxY, maxZ] = max
+
+    let anchor = block.getPos()
+    const anchorArray = blockEntity.data.getIntArray("anchor")
+    if (anchorArray.length >= 3)
+      anchor = new BlockPos(anchorArray[0], anchorArray[1], anchorArray[2])
+    else
+      player.tell(Text.translate("block.ptdye.animation_block").append(": ").green().append(Component.ofString("No anchor set, using position of animation block").white()))
+
+    const [neighbours] = animation_findGluedNeighbours(block)
+    const nbt = Ptdye.utils.createContraptionNBTFrom(event.level, neighbours, anchor, AABB.of(minX, minY, minZ, maxX, maxY, maxZ))
+    console.log(nbt)
```

Comment:
console.log here

Reply:
Thanks for pointing it out; I've actually kept this in on purpose. So, when I do the 2nd part of the animation system I can validate the contraptions. Plus it can be debugged/validated by others if they wish to test the code by building something and then checking that it produces the correct contraption that they built